### PR TITLE
5

### PR DESCRIPTION
**Pull request by NacerShucks**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -81,6 +79,30 @@ describe('GET /api/articles/:article_id', () => {
...
+                expect(Object.keys(article)).toContain('article_id') 
+                expect(Object.keys(article)).toContain('topic') 
+                expect(Object.keys(article)).toContain('created_at') 
+                expect(Object.keys(article)).toContain('article_img_url') 
+                expect(Object.keys(article)).toContain('votes') 
+                expect(Object.keys(article)).toContain('comment_count') 
+                expect(Object.keys(article)).not.toContain('body')               
+            })
+        })
+    });
```
Comment (nc-kev-morel) [inline __tests__/app.test.js:104]: Nearly there. We just need a test to check the order of these articles is correct

Comment (NacerShucks) [reply __tests__/app.test.js:104]: yeah i fully got too excited that the sql query worked and thought i was done lol

File: controllers/articles.controller.js [hunk 2]
```diff
@@ -1,12 +1,23 @@
-const { selectArticleById } = require("../modules/articles.modules")
+const { selectArticleById, selectArticles } = require("../modules/articles.modules")
 
 exports.getArticle = (req, res, next) => {
```
Comment (nc-hannah-burks) [inline controllers/articles.controller.js]: We have over complicated this by combining getAllArticles and get ArticleById - we want these to be separate ideally, as it makes our code more organised and easy to read.

File: app.js [hunk 3]
```diff
@@ -11,6 +11,8 @@ app.get('/api/topics', getTopics)
 
 app.get('/api', getEndpoints)
 
+app.get('/api/articles/', getArticle)
```
Comment (nc-hannah-burks) [inline app.js]: As we are wanting multiple articles, the controller being called getArticle (singular) does not make sense. Therefore, we need separate controllers and models for the two endpoints ( /api/articles/ and '/api/articles/:article_id')

File: modules/articles.modules.js [hunk 4]
```diff
@@ -9,4 +9,24 @@ exports.selectArticleById = (id) => {
...
+                                COUNT(comments.article_id) AS comment_count
+                                FROM articles
+                                LEFT JOIN comments 
+                                ON articles.article_id = comments.article_id 
+                                GROUP BY articles.article_id
+                                ORDER BY articles.created_at DESC`
+    return db.query(queryString)
+    .then(({rows}) => {
+        return rows
+    })
```
Comment (nc-hannah-burks) [inline modules/articles.modules.js]: This file/folder should be models, not modules 🙂

File: modules/articles.modules.js [hunk 5]
```diff
@@ -9,4 +9,24 @@ exports.selectArticleById = (id) => {
...
+                                article_img_url,
+                                COUNT(comments.article_id) AS comment_count
+                                FROM articles
+                                LEFT JOIN comments 
+                                ON articles.article_id = comments.article_id 
+                                GROUP BY articles.article_id
+                                ORDER BY articles.created_at DESC`
+    return db.query(queryString)
+    .then(({rows}) => {
+        return rows
```
Comment (nc-hannah-burks) [inline modules/articles.modules.js]: Again, we should have two models for the two end points. When returning just a singular article, we want to send back an object, for multiple articles we want to send back an array

File: __tests__/app.test.js [hunk 6]
```diff
@@ -81,7 +79,30 @@ describe('GET /api/articles/:article_id', () => {
...
+            body.forEach((article) => {
+                expect(Object.keys(article)).toContain('author') 
+                expect(Object.keys(article)).toContain('title') 
+                expect(Object.keys(article)).toContain('article_id') 
+                expect(Object.keys(article)).toContain('topic') 
+                expect(Object.keys(article)).toContain('created_at') 
+                expect(Object.keys(article)).toContain('article_img_url') 
+                expect(Object.keys(article)).toContain('votes') 
+                expect(Object.keys(article)).toContain('comment_count') 
+                expect(Object.keys(article)).not.toContain('body')
```
Comment (nc-hannah-burks) [inline __tests__/app.test.js:98]: Nice checking that body is not included here!

File: __tests__/app.test.js [hunk 7]
```diff
@@ -81,7 +79,30 @@ describe('GET /api/articles/:article_id', () => {
...
+            expect(body).toHaveLength(13)
+            body.forEach((article) => {
+                expect(Object.keys(article)).toContain('author') 
+                expect(Object.keys(article)).toContain('title') 
+                expect(Object.keys(article)).toContain('article_id') 
+                expect(Object.keys(article)).toContain('topic') 
+                expect(Object.keys(article)).toContain('created_at') 
+                expect(Object.keys(article)).toContain('article_img_url') 
+                expect(Object.keys(article)).toContain('votes') 
+                expect(Object.keys(article)).toContain('comment_count')
```
Comment (nc-hannah-burks) [inline __tests__/app.test.js:97]: You may also want to test the comment count is correct on one of the objects returned

**Review by nc-hannah-burks — COMMENTED** [review]
A good start, but let's separate those two end points out. Please resub once done.

**Review by ghost — APPROVED** [review]
Nice refactor, well done.

**Pull request merged**